src/Ocelot/Multiplexer/MultiplexingMiddleware.cs

```diff
@@ -255,5 +255,27 @@ protected virtual Task MapAsync(HttpContext httpContext, Route route, List<HttpC
 
         var aggregator = _factory.Get(route);
         return aggregator.Aggregate(route, httpContext, contexts);
+    }
+
+    protected virtual async Task<Stream> CopyBufferToTargetRequestAsync(HttpContext source)
+    {
+        source.Request.EnableBuffering();

```

Comment:
This `EnableBuffering()` extension method comes from the AspNet framework: https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.httprequestrewindextensions.enablebuffering?view=aspnetcore-8.0

This seems to be the recommended way to enable native capabilities: https://devblogs.microsoft.com/dotnet/re-reading-asp-net-core-request-bodies-with-enablebuffering/

Feel free to discuss this solution here, I think it's pretty elegant.

Reply:
We're aware of this fantastic ASP.NET feature 😄. However, as a team, we've opted not to use `HttpClient` and instead utilize the new `HttpMessageInvoker` pooling mechanism to enhance our heavy streaming capabilities.
For multiplexing, we should employ local buffering and replicate the body multiple times into downstreams.

Reply:
### The Q
>Ensure the request [Body](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.httprequest.body?view=aspnetcore-8.0#microsoft-aspnetcore-http-httprequest-body) **can be read multiple times**. Normally buffers request bodies in memory; writes requests larger than 30K bytes to disk.

>Ensure the request body can be read multiple times.

This is exactly what we need for multiplexing scenario!

Reply:
@raman-m thanks for the input, it is enabled on ASP.NET Request, not on HttpClient.

Where would you put the settings for this buffering? I thought it would be preferable to let it by default at first because this would not be enabled for requests that are not multiplexed, thus being a bit confusing.

Also buffering was not part of the story which is why I did the strict minimum buffering. Considering the remark on the previous PR on overstepping the boundaries of the bug, I think it would be best to postpone this (interesting) addition/customization to a more global buffering story.